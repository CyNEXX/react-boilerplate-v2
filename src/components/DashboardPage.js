import React from 'react';

const DashboardPage = () => (
    <div className='d-flex justify-content-between flex-column'>
        Dashboard page
    </div>
);

export default DashboardPage;