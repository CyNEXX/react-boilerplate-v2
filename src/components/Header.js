import React from 'react';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

import Button from 'react-bootstrap/Button';
import { Link } from 'react-router-dom';

export const Header = ({ startLogout, displayName, photoURL }) => {
    return (
        <div className='bg-dark text-light'>
            <div className='content-container d-flex justify-content-between py-2 px-3'>
                <Link className='font-weight-bold text-light h2 mb-0' to="/dashboard">boilerplate</Link>
                <div className="justify-content-end text-light">
                    <span>Signed in as: </span><span className='font-weight-bold'>{displayName}</span>
                    <img className='profile_pic mx-3' src={photoURL} alt='Profile picture'></img>
                    <Button type='button' id='signout' className='btn btn-dark btn-outline-light' onClick={startLogout}>Sign Out</Button>
                </div>
            </div>
        </div>
    )
};

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});


export default connect(undefined, mapDispatchToProps)(Header);