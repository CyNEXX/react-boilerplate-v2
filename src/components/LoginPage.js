import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className='main-bg d-flex'>
        <div className='d-flex'>
            <div className='login-box mr-auto ml-auto mt-auto mb-auto text-center px-0 py-4 blur'>
                <h1 className='login-title m-0 mb-3 font-weight-bold motto'>Boilerplate</h1>
                <h6 className='motto font-italic'>tag gline</h6>
                <button className='btn btn-primary mt-2' onClick={startLogin}>Log In with Google</button>
            </div>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
