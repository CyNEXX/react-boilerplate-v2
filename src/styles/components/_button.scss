.bigbutton {
    @include (max-width: 1199.98px) {
        height: 7rem !important;
        vertical-align: middle !important;
    }
}
