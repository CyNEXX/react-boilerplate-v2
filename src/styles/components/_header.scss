.profile_pic {
    height: 2rem;
}