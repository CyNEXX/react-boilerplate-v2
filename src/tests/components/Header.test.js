// react-test-renderer
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

test('Should render Header correctly', () => {
    const wrapper = shallow(<Header startSignout={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogout', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('#signout').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});