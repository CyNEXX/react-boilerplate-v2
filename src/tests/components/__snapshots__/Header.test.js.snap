// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div
  className="bg-dark text-light"
>
  <div
    className="content-container d-flex justify-content-between py-2 px-3"
  >
    <Link
      className="font-weight-bold text-light h2 mb-0"
      to="/dashboard"
    >
      boilerplate
    </Link>
    <div
      className="justify-content-end text-light"
    >
      <span>
        Signed in as: 
      </span>
      <span
        className="font-weight-bold"
      />
      <img
        alt="Profile picture"
        className="profile_pic mx-3"
      />
      <Button
        active={false}
        className="btn btn-dark btn-outline-light"
        disabled={false}
        id="signout"
        type="button"
        variant="primary"
      >
        Sign Out
      </Button>
    </div>
  </div>
</div>
`;
